kafka/channel/pkg/reconciler/controller/kafkachannel.go

```diff
@@ -370,6 +388,52 @@ func (r *Reconciler) reconcile(ctx context.Context, kc *v1alpha1.KafkaChannel) e
 	return nil
 }
 
+func (r *Reconciler) reconcileDispatcher(ctx context.Context, dispatcherNamespace string, kc *v1alpha1.KafkaChannel) (*appsv1.Deployment, error) {
+	d, err := r.deploymentLister.Deployments(dispatcherNamespace).Get(dispatcherDeploymentName)
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			args := resources.DispatcherArgs{
+				DispatcherNamespace: dispatcherNamespace,
+				Image:               r.dispatcherImage,
+			}
+			expected := resources.MakeDispatcher(args)
+			d, err := r.KubeClientSet.AppsV1().Deployments(dispatcherNamespace).Create(expected)
+			msg := "Dispatcher Deployment created"
+			if err != nil {
+				msg = fmt.Sprintf("not created, error: %v", err)
+			}
+			r.Recorder.Eventf(kc, corev1.EventTypeNormal, dispatcherDeploymentCreated, "%s", msg)
```

Comment:
Do we still want to record a `dispatcherDeploymentCreated` event in the error case? Wouldn't we want a `dispatcherDeploymentFailed` event passed?